hqx/src/common.rs

```diff
@@ -109,3 +109,42 @@ pub fn interp10(c1: u32, c2: u32, c3: u32) -> u32 {
     //(c1*14+c2+c3)/16;
     interpolate3(c1, 14, c2, 1, c3, 1, 4)
 }
+
+// A wrapper that allows any indexing panics to point to the same location,
+// significantly reducing size of the generated code.
+#[repr(transparent)]
+pub struct Dst([u32]);
+
+impl Dst {
+    pub fn from_mut(dst: &mut [u32]) -> &mut Self {
```

Comment:
Can we make this constructor unsafe? Because otherwise the fact that misuse is UB is not exposed in the type's API at all.

Reply:
Ah yeah sure. I initially wrote version without `unsafe` indexing, so this was safe to construct, but I agree now it should be unsafe.

Reply:
Done & renamed to `UncheckedDst` to make it a bit more clear what's going on.